Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -150,16 +150,35 @@ void TextureCache::Cleanup(int _frameCount)
 		if (iter->second->frameCount == FRAMECOUNT_INVALID)
 		{
 			iter->second->frameCount = _frameCount;
-		}
-		if (_frameCount > TEXTURE_KILL_THRESHOLD + iter->second->frameCount &&
-		    // EFB copies living on the host GPU are unrecoverable and thus shouldn't be deleted
-		    !iter->second->IsEfbCopy())
-		{
-			iter = FreeTexture(iter);
+			++iter;
 		}
 		else
 		{
-			++iter;
+			if (_frameCount > TEXTURE_KILL_THRESHOLD + iter->second->frameCount)
+			{
+				if (iter->second->IsEfbCopy())
+				{
+					// Only remove EFB copies when they wouldn't be used anymore(changed hash), because EFB copies living on the
+					// host GPU are unrecoverable. Perform this check only every TEXTURE_KILL_THRESHOLD for performance reasons
+					if ((_frameCount - iter->second->frameCount) % TEXTURE_KILL_THRESHOLD == 1 &&
```

Comment:
Are you sure about the order?
_frameCount increase by one per frame. So a frame later, this difference is 1.

"iter->second->frameCount - _frameCount"